src/workerManager.ts

```diff
@@ -23,12 +23,18 @@ export class WorkerManager {
 	private _worker: monaco.editor.MonacoWebWorker<TypeScriptWorker>;
 	private _client: Promise<TypeScriptWorker>;
 
-	constructor(defaults: LanguageServiceDefaultsImpl) {
+	private _stopWhenIdleFor: number;
+
+	constructor(defaults: LanguageServiceDefaultsImpl, stopWhenIdleFor?: number) {
 		this._defaults = defaults;
 		this._worker = null;
-		this._idleCheckInterval = setInterval(() => this._checkIfIdle(), 30 * 1000);
 		this._lastUsedTime = 0;
 		this._configChangeListener = this._defaults.onDidChange(() => this._stopWorker());
+		this._stopWhenIdleFor = (stopWhenIdleFor || stopWhenIdleFor === 0) ? stopWhenIdleFor : STOP_WHEN_IDLE_FOR;

```

Comment:
if we pass 0 as param this will make the worker persistent. I'm not sure if you prefer this syntax or the full if-else.